examples/build_customized_agent.py

```diff
@@ -0,0 +1,136 @@
+        return a + b
+    # test cases
+    print(add(1, 2))
+    print(add(3, 4))
+    ```
+    your code:
+    """
+
+    def __init__(self, name="SimpleWriteCode", context=None, llm=None):
+        super().__init__(name, context, llm)
+    
+    async def run(self, instruction: str):
+
+        prompt = self.PROMPT_TEMPLATE.format(instruction=instruction)
+        # logger.info(prompt)
+
+        rsp = await self._aask(prompt)
+
+        code_text = SimpleWriteCode.parse_code(rsp)
+
+        return code_text
+    
+    @staticmethod
+    def parse_code(rsp):
+        pattern = r'```python(.*)```'
+        match = re.search(pattern, rsp, re.DOTALL)
+        code_text = match.group(1) if match else rsp
+        return code_text
+
+class SimpleRunCode(Action):
+    def __init__(self, name="SimpleRunCode", context=None, llm=None):
+        super().__init__(name, context, llm)
+    
+    async def run(self, code_text: str):
+        result = subprocess.run(["python3", "-c", code_text], capture_output=True, text=True)
+        code_result = result.stdout
+        # exec(code_text)
+        logger.info(f"{code_result=}")
+        return code_result
+
+class SimpleCoder(Role):
+    def __init__(
+        self,
+        name: str = "Alice",
+        profile: str = "SimpleCoder",
+        **kwargs,
+    ):
+        super().__init__(name, profile, **kwargs)
+        self._init_actions([SimpleWriteCode])
+
+    async def _act(self) -> Message:
+        logger.info(f"{self._setting}: ready to {self._rc.todo}")
+        todo = self._rc.todo
+        msg = self._rc.memory.get()[-1]
+
+        instruction = msg.content
+        code_text = await SimpleWriteCode().run(instruction)
+        msg = Message(content=code_text, role=self.profile, cause_by=todo)
+
+        return msg
+
+class RunnableCoder(Role):
+    def __init__(
+        self,
+        name: str = "Alice",
+        profile: str = "RunnableCoder",
+        **kwargs,
+    ):
+        super().__init__(name, profile, **kwargs)
+        self._init_actions([SimpleWriteCode, SimpleRunCode])
+
+    async def _think(self) -> None:
+        if self._rc.todo is None:
+            self._set_state(0)
+            return
+
+        if self._rc.state + 1 < len(self._states):
+            self._set_state(self._rc.state + 1)
+        else:
+            self._rc.todo = None
+
+    async def _act(self) -> Message:
+        logger.info(f"{self._setting}: ready to {self._rc.todo}")
+        todo = self._rc.todo
+        msg = self._rc.memory.get()[-1]
+
+        if isinstance(todo, SimpleWriteCode):
+            instruction = msg.content
+            code_text = await SimpleWriteCode().run(instruction)
+            msg = Message(content=code_text, role=self.profile, cause_by=todo)
+        
+        elif isinstance(todo, SimpleRunCode):
+            code_text = msg.content
+            rsp = await SimpleRunCode().run(code_text)
+            msg = Message(content=rsp, role=self.profile, cause_by=todo)
```

Comment:
put `msg = Message(content=rsp, role=self.profile, cause_by=todo)` outside if-else is better.